### DISCUSSION TITLE
Errors (Index was out of range) with PredictPose node using Bonsai-DLC

### DISCUSSION THREAD

**Discussion by aikasaito** [bonsai-rx/bonsai#1478] [Q&A] [answered]
Hi all,

I am trying to track animal’s body parts and get their xy coordinates using Bonsai-DLC. 

When I selected a video for the FileCapture and connected it to the PredictPose, I got a run-time error(Index was out of range. Must be non-negative and less than the size of the collection. Parameter name: index) (see Fig.1). When I used the CameraCapture or VideoCaptureDevice instead of the FileCapture, I got the same error.

On the other hand, when I used PredictPoseMaps, a video with labeled body parts was outputted, suggesting that the tracking itself seems working. However, when I connected the GetBodyPart to the PredictPoseMaps, I got another error(Build Error. No method overload found for the given arguments) (see Fig.2).

I am hopeful that someone here will know what the source of these errors is so that I can start using DLC in Bonsai.

Thanks for all your help!
Aika

Fig.1
![Fig1_PredictPose](https://github.com/bonsai-rx/bonsai/assets/49588505/883cae33-87b1-4dab-84f3-a95b6bc5f2ff)

Fig.2
![Fig2_GetBodyPart](https://github.com/bonsai-rx/bonsai/assets/49588505/28f3b619-556f-4d19-ab1c-d3adaefbd52a)

**Comment by bruno-f-cruz** [comment]
Can you share your network config file?

**Reply by aikasaito** [reply]
Hi Bruno,

Thank you for your quick reply!
Yes, of course. Here is the config file.

[config.yaml.zip](https://github.com/bonsai-rx/bonsai/files/12094274/config.yaml.zip)

Best regards,
Aika

**Reply by bruno-f-cruz** [reply]
Can you attach the .pb network? I don't see anything obviously wrong with the config...

**Reply by aikasaito** [reply]
Yes, of course. Here is the .pb file.
https://drive.google.com/file/d/1Sffu1TRqsdeDkP3ExCfrZNVvGs6dRkOv/view?usp=sharing

**Reply by bruno-f-cruz** [reply]
What version of dlc did you use to train and export the network? What routine did you use to export the .pb file?

**Comment by bruno-f-cruz** [comment, marked as answer]
Ok I think I found out the problem, but I am not sure what is going on with deeplabcut.

Bonsai uses the following entry in the config file to find bodyparts:
https://github.com/DeepLabCut/DeepLabCut/blob/d2ef2c04a62dedf30942acffc8dec333d173c42a/deeplabcut/pose_estimation_tensorflow/superanimal_configs/superquadruped.yaml#L41

For some reason your config file does not have this entry. You can fix it by duplicating the `bodyparts` entry and rename it to `all_joints_names`.

I am still curious as to why this is no longer in the config file, I wonder if the devs from DLC changed it recently perhaps?
If you find out what is going on, let us know, we are open to update the package if needed :)

**Reply by aikasaito** [reply]
Thank you so much!!

After adding the code following your advice, now I can use PredictPose and obtain xy coordinates for body parts.
I’m sorry but as for the reason why ‘all_joints_names’ was no longer in my config file, I’m not sure either. The config file I shared was, as far as I remember, created using deeplabcut(ver.2.2b8).

Anyway, your advice helped me a lot. Thank you!!
